deepchecks/nlp/checks/data_integrity/special_characters.py

```diff
@@ -0,0 +1,180 @@
+        self.random_state = random_state
+        self.max_text_length_for_display = max_text_length_for_display
+
+    @property
+    def _text_normalization_kwargs(self):
+        return {
+            'ignore_case': self.ignore_case,
+            'ignore_whitespace': self.ignore_whitespace,
+            'normalize_uni': self.normalize_unicode,
+            'remove_punct': self.remove_punctuation,
+            'remove_stops': self.remove_stopwords,
+        }
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind).sample(self.n_samples, random_state=self.random_state)
+        dataset = t.cast(TextData, dataset)
+        samples = dataset.text
+        n_of_samples = len(samples)
+
+        if n_of_samples == 0:
+            raise DeepchecksValueError('Dataset cannot be empty')
+
+        data: t.Dict[str, SpecialCharacterInfo] = {}
+
+        for char in self.special_characters:
+            for idx, sample in zip(dataset.get_original_text_indexes(), samples):
+                if char in sample:
+                    data[char] = data.get(char, {
+                        'samples_ids': [],
+                        'text_example': sample,
+                        'percent_of_samples': 0
+                    })
+                    data[char]['samples_ids'].append(idx)
+
+        for char, info in data.items():
+            info['percent_of_samples'] = len(info['samples_ids']) / n_of_samples
+
+        if context.with_display is False or len(data) == 0:
+            return CheckResult(value=data)
+
+        display_table = pd.DataFrame(
+            index=range(len(data)),
+            columns=['Special Character', '% of Samples With Character', 'Instances', 'Text Example'],
+            data=[
+                [char,
+                 values['percent_of_samples'],
+                 format_list(values['samples_ids']),
+                 truncate_string(values['text_example'], self.max_text_length_for_display)]
+                for char, values in data.items()
+            ],
+        )
+        display_table = (
+            display_table.sort_values(by=['% of Samples With Character'], ascending=False)
+            .reset_index(drop=True)
+            .set_index(['Special Character'])
+        )
+        if self.n_most_common > display_table.shape[0]:
+            message = ''
+        else:
+            message = (
+                f'Showing only the top {self.n_most_common} most common characters, '
+                'you can change it using n_most_common param'
+            )
+        return CheckResult(
+            value=data,
+            display=[message, display_table.iloc[:self.n_most_common]]
+        )
+
+    # TODO:
+    # is default max_ratio good?
+    # are method and condition names good?
+    def add_condition_ratio_of_special_characters_less_or_equal(self: Self, max_ratio: float = 0.05) -> Self:
```

Comment:
I think an additional condition that perhaps is better as a default if on the total % of samples that have any of those special characters.

Reply:
added additional condition